src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -50,6 +50,9 @@
     "The output includes credentials that you must protect. Be sure that you do not include these credentials in "
     "your code or check the credentials into your source control. For more information, see https://aka.ms/azadsp-cli")
 
+SCOPE_WARNING = "In a future release, --scope argument will become required for creating a role assignment. " \
```

Comment:
Similar to #20965, we can't give an exact version yet.

Reply:
@jiasli , but can't we at least give the anticipated breaking change (BK) window as are BKs are moving to a bi-annual schedule?

Reply:
this should also be consistent with the content on line 688.
We can mention something like "breaking change of Fall of 2023".